packages/ckeditor5-dev-env/bin/save-revision.js

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
+
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const branch = 'master'; // process.env.TRAVIS_BRANCH;
+
+if ( branch !== 'master' ) {
+	process.exit();
+}
+
+const path = require( 'path' );
+const { tools } = require( '@ckeditor/ckeditor5-dev-utils' );
+const execOnDependencies = require( '@ckeditor/ckeditor5-dev-env/lib/utils/exec-on-dependencies' );
+
+const mainBranch = `${ branch }-revisions`;
+
+// Clone the repository.
+// jscs:disable maximumLineLength
+exec( `git clone -b ${ mainBranch } https://${ process.env.GITHUB_USER }:${ process.env.GITHUB_PASSWORD }@github.com/ckeditor/ckeditor5.git` );
+// jscs:enable maximumLineLength
+
+const cwd = path.join( process.cwd(), 'ckeditor5' );
+const mgitJsonPath = path.join( cwd, 'mgit.json' );
+const mgitJson = require( mgitJsonPath );
+const depsHashes = {};
+
+const execDepsOptions = {
+	cwd,
+	workspace: mgitJson.packages
+};
+
+const previousPwd = exec( 'pwd' ).trim();
+
+// Change current dir to cloned repository.
+process.chdir( cwd );
+
+// Install Mgit
+exec( 'npm install -g mgit2' );
+
+// Install dependencies.
+// jscs:disable maximumLineLength
+exec( `mgit bootstrap --recursive --repository-resolver=${ previousPwd }/node_modules/@ckeditor/ckeditor5-dev-tests/lib/git-https-resolver.js` );
+// jscs:enable maximumLineLength
+
+// Checkout out each package to master.
+exec( `mgit exec "git checkout ${ branch }"` );
+
+// Get hashes from all dependencies
+execOnDependencies( execDepsOptions, functionToExecute )
+	.then( () => {
+		const repository = process.env.TRAVIS_REPO_SLUG;
+
+		tools.updateJSONFile( mgitJsonPath, ( json ) => {
+			for ( const packageName of Object.keys( depsHashes ) ) {
+				let packageRepo = packageName;
+
+				if ( packageRepo.startsWith( '@' ) ) {
+					packageRepo = packageName.slice( 1 );
+				}
+
+				json.dependencies[ packageName ] = packageRepo + '#' + depsHashes[ packageName ];
+			}
+
+			json.dependencies[ repository ] = repository + '#' + process.env.TRAVIS_COMMIT;
```

Comment:
In order to avoid invalid keys in [`mgit.json`](https://github.com/ckeditor/ckeditor5/blob/master-revisions/mgit.json#L22),`repository` in `json. dependencies ` should starting with '@'.